addons/mail/models/mail_thread.py

```diff
@@ -721,78 +765,71 @@ def message_route_verify(self, message, message_dict, route,
+                thread_id = None
             elif not hasattr(record_set, 'message_update'):
-                self._routing_warn(_('model %s does not accept document update') % model, _('skipping'), message_id, route, assert_model)
-                return False
+                self._routing_warn(_('reply to model %s that does not accept document update, fall back on document creation') % model, message_id, route, False)
+                thread_id = None
 
         # New Document: check model accepts the mailgateway
         if not thread_id and model and not hasattr(record_set, 'message_new'):
-            self._routing_warn(_('model %s does not accept document creation') % model, _('skipping'), message_id, route, assert_model)
-            return False
+            self._routing_warn(_('model %s does not accept document creation') % model, message_id, route, raise_exception)
+            return ()
 
         # Update message author if asked. We do it now because we need it for aliases (contact settings)
-        if not author_id and update_author:
+        if not author_id:
             authors = self._mail_find_partner_from_emails([email_from], records=record_set)
             if authors:
                 message_dict['author_id'] = authors[0].id
 
         # Alias: check alias_contact settings
         if alias:
-            obj = None
             if thread_id:
                 obj = record_set[0]
             elif alias.alias_parent_model_id and alias.alias_parent_thread_id:
                 obj = self.env[alias.alias_parent_model_id.model].browse(alias.alias_parent_thread_id)
-            elif model:
+            else:
                 obj = self.env[model]
             if hasattr(obj, '_alias_check_contact'):
                 check_result = obj._alias_check_contact(message, message_dict, alias)
             else:
                 check_result = self.env['mail.alias.mixin']._alias_check_contact_on_record(obj, message, message_dict, alias)
             if check_result is not True:
-                self._routing_warn(_('alias %s: %s') % (alias.alias_name, check_result.get('error_message', _('unknown error'))), _('skipping'), message_id, route, False)
+                self._routing_warn(_('alias %s: %s') % (alias.alias_name, check_result.get('error_message', _('unknown error'))), message_id, route, False)
                 self._routing_create_bounce_email(email_from, check_result.get('error_template', _generic_bounce_body_html), message)
                 return False
 
-        if not model and not thread_id and not alias and not allow_private:
-            return False
+        return (model, thread_id, route[2], route[3], route[4])
 
-        return (model, thread_id, route[2], route[3], None if drop_alias else route[4])
+    @api.model
+    def _routing_reset_bounce(self, email_message, message_dict):
```

Comment:
I agree there is some cleaning to bo. However I would keep a message_reset_bounce method allowing to implement some custom behavior based on bounce reset. We had some specs somewhere to further improve bounce management and its reset, it could land in that method.

However I will try to lessen call chain because you are right it is a bit messy for such a small final result :) .

Reply:
The call chain is not a real problem for me actually, It is more the missing domain ('message_bounce', '>', 0). ;)

Reply:
Done !